python/pyarrow/table.pxi

```diff
@@ -766,6 +766,40 @@ cdef class Table(_PandasConvertible):
         columns = [col for col in self.columns]
         return _reconstruct_table, (columns, self.schema)
 
+    def __getitem__(self, key):
+        if isinstance(key, slice):
+            return _normalize_slice(self, key)
+        else:
+            return self.column(key)
```

Comment:
Note that this is a (unintentional?) behaviour change (by using `self.column(i)` instead of `self._column(i)`), because is now accepts both string names as integer index (up to now, the `table.column(i)` method accepted both, but `__getitem__` only integers)